lib/Jit/EEMemoryManager.cpp

```diff
@@ -130,13 +181,40 @@ void EEMemoryManager::registerEHFrames(uint8_t *Addr, uint64_t LoadAddr,
   // ColdCodeBlock, i.e. separated code is not supported.
   assert(this->ColdCodeBlock == 0 && "ColdCodeBlock must be zero");
 
-  // Assume this unwind covers the entire method. Later when
-  // we have multiple unwind regions we'll need something more clever.
-  uint32_t StartOffset = 0;
-  uint32_t EndOffset = this->Context->HotCodeSize;
-  this->Context->JitInfo->allocUnwindInfo(
-      this->HotCodeBlock, nullptr, StartOffset, EndOffset, Size,
-      (BYTE *)LoadAddr, CorJitFuncKind::CORJIT_FUNC_ROOT);
+  uint32_t *DataPtr = reinterpret_cast<uint32_t *>(Addr);
+  CorJitFuncKind FuncKind = CorJitFuncKind::CORJIT_FUNC_ROOT;
+  uint32_t IsLast;
+  do {
```

Comment:
If we ever had chained unwind in the main root (eg from shrink wrap or separation) this would get confused, right? Is there any way for us to know if we see such cases here?